Userland/Libraries/LibWeb/TreeNode.h

```diff
@@ -137,12 +137,30 @@ class TreeNode : public Weakable<T> {
         return node;
     }
 
-    const T* next_in_pre_order() const
+    T const* next_in_pre_order() const
     {
         return const_cast<TreeNode*>(this)->next_in_pre_order();
     }
 
-    bool is_before(const T& other) const
+    T* previous_in_pre_order()
+    {
+        T* node;
+        if ((node = previous_sibling())) {
```

Comment:
```suggestion
        if (auto node = previous_sibling()) {
```
the node is only local to this `if` anyway.